private/bufpkg/bufapimodule/module_reader.go

```diff
@@ -75,7 +75,12 @@ func (m *moduleReader) GetModule(ctx context.Context, modulePin bufmoduleref.Mod
 		if err != nil {
 			return nil, err
 		}
-		return bufmodule.NewModuleForBucket(ctx, bucket, identityAndCommitOpt)
+		return bufmodule.NewModuleForBucket(
+			ctx,
+			bucket,
+			identityAndCommitOpt,
+			bufmodule.ModuleWithManifestAndBlobs(bucket.Manifest(), bucket.Blobs()),
```

Comment:
This is important - if we have manifest/blobs, we need to preserve them in the buf module. This allows us to store them to the cache with all the metadata.